src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/CVService/CVObjectDetectionResult.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Text;
+using RegressionGames.StateRecorder.JsonConverters;
+using UnityEngine;
+
+namespace RegressionGames.StateRecorder.BotSegments.Models.CVService
+{
+    public class CVObjectDetectionResult
+    {
+        public string text;
```

Comment:
This is very wrong. Fixing now, however the fact I missed it is worrying and potentially calls for a hardening.

Basically this is what happened:
1) I assumed compiler would complain about a mismatch between result json and string, when it didn't I didn't double check. This is not the case, as we use 'TryGetValue' to fill in the results, which does its best to load results but doesn't enforce a match.
```c#
public bool TryGetValue(TKey key, out TValue value)
    {
      int entry = this.FindEntry(key);
      if (entry >= 0)
      {
        value = this._entries[entry].value;
        return true;
      }
      value = default (TValue);
      return false;
    }
```
2) The criteria only cares about existence of a match, not whether the data returned is correct (well because we don't have ground truth). So even if the type was very wrong, the behavior worked as expected. This would have fallen over very hard at the action phase.

It is likely there is a reason why this is the way it is, so I am adding a link to the future [REG-1928](https://linear.app/regression-games/issue/REG-1928/refactor-cv-object-detection-evaluator) to discuss hardening

Reply:
Good catch!